packages/lab/src/dropdown-next/DropdownNext.tsx

```diff
@@ -0,0 +1,250 @@
+      highlightedItem,
+      getListItems,
+      portalProps,
+    } = useDropdownNext({
+      source,
+      placement,
+      listProps,
+      openControlProp,
+    });
+
+    const { open, floating, reference, getDropdownNextProps } = portalProps;
+    const {
+      focusHandler,
+      keyDownHandler,
+      blurHandler,
+      mouseOverHandler,
+      mouseDownHandler,
+    } = handlers;
+
+    const triggerRef = useForkRef(
+      ref,
+      reference
+    ) as MutableRefObject<HTMLButtonElement>;
+
+    const floatingListRef = useForkRef(
+      listRef,
+      floating
+    ) as MutableRefObject<HTMLUListElement>;
+
+    const getIcon = () => {
+      const iconClassName = clsx(
+        withBaseName("icon"),
+        {
+          [withBaseName("disabled")]: disabled,
+        },
+        className
+      );
+
+      return readOnly ? (
+        ""
+      ) : open ? (
+        <ChevronUpIcon className={iconClassName} />
+      ) : (
+        <ChevronDownIcon className={iconClassName} />
+      );
+    };
+
+    const handleFocus = (event: FocusEvent<HTMLElement>) => {
+      if (disabled || readOnly) return;
+      focusHandler(event);
+      onFocus?.(event);
+    };
+
+    const handleKeyDown = (event: KeyboardEvent<HTMLButtonElement>) => {
+      if (disabled || readOnly) return;
+      keyDownHandler(event);
+      onKeyDown?.(event);
+    };
+
+    const handleBlur = (event: FocusEvent<HTMLElement>) => {
+      blurHandler();
+      onBlur?.(event);
+    };
+
+    const handleMouseOver = (event: MouseEvent<HTMLElement>) => {
+      mouseOverHandler();
+      onMouseOver?.(event);
+    };
+
+    const handleMouseDown = (event: MouseEvent<HTMLElement>) => {
+      if (disabled || readOnly) return;
+      mouseDownHandler();
+      onMouseDown?.(event);
+    };
+
+    return (
+      <div className={clsx(withBaseName(), className)}>
+        <button
+          id={dropdownId}
+          disabled={disabled}
+          onFocus={handleFocus}
+          onKeyDown={handleKeyDown}
+          onMouseOver={handleMouseOver}
+          onMouseDown={handleMouseDown}
+          onBlur={handleBlur}
+          value={selectedItem}
+          className={clsx(
+            withBaseName("button"),
+            withBaseName(variant),
+            {
+              [withBaseName("disabled")]: disabled,
+              [withBaseName("readOnly")]: readOnly,
+            },
+            className
+          )}
+          role="combobox"
+          aria-haspopup="listbox"
+          aria-expanded={open}
+          aria-activedescendant={disabled ? undefined : activeDescendant}
+          tabIndex={disabled ? -1 : 0}
+          aria-owns={listId}
+          aria-controls={listId}
+          aria-disabled={disabled}
+          style={dropdownStyle}
+          {...restProps}
+          ref={triggerRef}
+        >
+          <span className={clsx(withBaseName("buttonText"), className)}>
+            {selectedItem}
+          </span>
+          {getIcon()}
```

Comment:
Not really seen this before - personally think the below is more readable. Would like to hear what you think


`<span
            className={clsx(
              withBaseName("icon"),
              { [withBaseName("disabled")]: disabled },
              className
            )}
          >
            {!readOnly && open ? <ChevronUpIcon /> : <ChevronDownIcon />}
          </span>`

Reply:
I had this previously but couldn't seem to get the alignment of Icon to work if it's wrapped in a <span>, let me give this another go and see! :D